File: daily/Arrays/136-Single-number.py
from collections import defaultdict
from typing import List

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        map_num = defaultdict(int)
        for num in nums:
           map_num[num] += 1

        for num in map_num:
            if map_num[num] != 2:
                return num
        return 0

File: daily/Arrays/136-Single-number.go
package daily

func singleNumber(nums []int) int {
	map_num := make(map[int]int)
	for _, num := range nums {
		map_num[num]++
	}

	for key, value := range map_num {
		if value != 2 {
			return key
		}
	}
	return 0
}
